ocfweb/account/print.py

```diff
@@ -0,0 +1,147 @@
+            )
+            assert entry is not None  # should never be none as we are passing a static string above
+            host_keys.add(
+                'ssh.ocf.berkeley.edu',
+                'ssh-rsa',
+                entry.key,
+            )
+
+            try:
+                ssh.connect(
+                    'ssh.ocf.berkeley.edu',
+                    username=username,
+                    password=password,
+                )
+            except AuthenticationException:
+                error = 'Authentication failed. Did you type the wrong username or password?'
+
+            if not error:
+                sftp = ssh.open_sftp()
+                try:
+                    folder = sftp.stat(PRINT_FOLDER)
+                    if folder.FLAG_PERMISSIONS != 0o700:
+                        sftp.chmod(PRINT_FOLDER, 0o700)
+                except FileNotFoundError:
+                    sftp.mkdir(PRINT_FOLDER, 0o700)
+                try:
+                    rid = uuid.uuid4().hex
+                    filename = f'{rid}-{file.name}'
+                    with sftp.open(os.path.join(PRINT_FOLDER, filename), 'wb+') as dest:
+                        for chunk in file.chunks():
+                            dest.write(chunk)
+                except OSError as e:
+                    error = 'Failed to open file in user home directory, ' + \
+                        f'please report this to help@ocf.berkeley.edu with the traceback\n{e}'
+                else:
+                    qr = qrcode.QRCode(
+                        version=1,
+                        box_size=10,
+                        border=5,
+                    )
+                    qr.add_data(f'{username}:{filename}:{double_or_single}')
+                    qr.make(fit=True)
+                    img = qr.make_image(fill='black', back_color='white')
+                    buff = BytesIO()
+                    img.save(buff, format='PNG')
+                    qr_b64 = b'data:image/png;base64,%b' % base64.b64encode(buff.getvalue())
+        return render(
+            request,
+            'account/print/qr.html',
+            {
+                'title': 'QR Code',
+                'qr_b64': qr_b64.decode('utf-8'),
+                'error': error,
+            },
+        )
+    else:
+        form = PrintForm(
+            initial={
+                'double_or_single': 'single',
+            },
+        )
+
+        return render(
+            request,
+            'account/print/index.html', {
+                'title': 'Print remotely',
+                'form': form,
+            },
+        )
+
+
+class PrintForm(Form):
+    username = forms.CharField(
+        label='OCF username',
```

Comment:
Can we grab this from the currently logged in user?

Reply:
No, I don't believe so. We have their calnet UID and that isn't one to one with an OCF account